lib/dealership.rb

```diff
@@ -0,0 +1,68 @@
+# require './lib/car'
+#
+#
+#
+class Dealership
+
+  attr_reader :name, :address, :inventory
+
+  def initialize(name, address)
+    @name = name
+    @address = address
+    @inventory = []
+    @total_val = 0
+    @inv_by_cost = []
+  end
+
+  def inventory_count
+    @inventory.count
+  end
+
+  def add_car(car)
+    @inventory << car
+  end
+
+  def has_inventory?
+    @inventory.count > 1
+  end
+
+  def cars_by_make(make)
+    by_make = @inventory.select do |car|
+      car.make == make
+    end
+    return by_make
+  end
+
+  def total_value
+    # total = 0
+    @inventory.each do |car|
+      @total_val += car.total_cost
+    end
+    return @total_val
+  end
```

Comment:
I would probably write this like so:

```
def total_value
  total = 0
  @inventory.each do |car|
    total += car.total_cost
  end
  total
```

You were really close with setting up a local variable for total!